### PR TITLE
Done three

### PR DESCRIPTION
**Pull request by gneral20**

### CODE REVIEW

**Comment by gneral20** [comment]
How difficult did you find this (0 - 10)? (0 very easy, 10 very difficult)
8

How long did it take?
3 Hours

Was there anything that you struggled with? If so, what?
Yes, Q3 and Q4

Is there anything that you would like some further information on?
built in functions and how to use them

Do you have any suggestions to improve this assignment?
More lab exercises on the same homework questions

File: MyAnswers.rb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  {
+      first_name: 'Jennifer',
+      last_name: 'Aniston',
+  }
+]
+
+upper_case_full_names = students.map do |x|
+
+    x[:first_name].upcase+ " " + x[:last_name].upcase
+
```
Comment (Salman778) [inline MyAnswers.rb:21]: Great!

File: MyAnswers.rb [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+          {
+              description: 'a kitten'
+          }
+      ]
+  }
+]
+
+first_order_for_each_user = users.map do |i|
+        i[:orders][0]
+      end
```
Comment (Salman778) [inline MyAnswers.rb:73]: Excellent!

File: MyAnswers.rb [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        coffee_sum = []
+        person[:transactions].map do |transaction|
+            if transaction[:type] == 'COFFEE'
+                coffee_sum.push(transaction[:amount])
+            end
+        end
+        
+        { 
+        name: person[:name] , 
+        coffee_average: coffee_sum.reduce(:+) / coffee_sum.length
```
Comment (Salman778) [inline MyAnswers.rb:148]: Wow!

**Comment by Salman778** [comment]
Good work! keep going!

**Pull request closed without merging**